src/app/meds-n-labels/activity-1/matchmeds.component.ts

```diff
@@ -0,0 +1,294 @@
+
+    /**
+     * Draws the Elemets on the Canvas
+     */
+    paintCanvasElements() {
+        for (let i = 0; i < this.numElements; i++) {
+            const spacing = this.gap * i;
+            this.setLineStyle('#466bc1', this.lineWidth);
+            this.cx.beginPath();
+            this.cx.arc(this.startPos + this.radius, spacing + this.radius + this.lineWidth, this.radius, 0, 2 * Math.PI);
+            this.cx.stroke();
+            this.cx.beginPath();
+            this.cx.arc(this.endPos , spacing + this.radius + this.lineWidth, this.radius, 0, 2 * Math.PI);
+            this.cx.stroke();
+        }
+    }
+
+    /**
+     * Set the Canvas Lines Style
+     * @param  {string} color   Color of the line
+     * @param  {number} lineWid Width of the line
+     */
+    setLineStyle(color, lineWid) {
+        this.cx.lineCap = 'round';
+        this.cx.strokeStyle = color;
+        this.cx.lineWidth = lineWid;
+    }
+
+    /**
+     * Checks the Size of the window and changes the size of the Canvas
+     * @param  {number} wid Width of the screen
+     */
+    checkWindowSize(wid) {
+        /**
+        * Default 75
+        * >400 - 100
+        * >768 - 200
+        * >992 - 250
+        */
+        if (wid > 400 && wid < 768) {
+            this.width = 100;
+        }else if (wid >= 768 && wid < 992) {
+            this.width = 200;
+        }else if (wid > 992) {
+            this.width = 250;
+        }
+    }
+
+    /**
+     * Gets the Top and Left coordiantes of the Canvas
+     */
+    updateCanvasCord() {
+        this.canvasTop = this.canvasElement.getBoundingClientRect().top;
+        this.canvasLeft = this.canvasElement.getBoundingClientRect().left;
+    }
+
+    /**
+     * The Event gets called when the scren size changes
+     * @param  {number} $event Contains screen size information
+     */
+    onResize($event) {
+        this.updateCanvasCord();
+        this.checkWindowSize($event.target.innerWidth);
+        this.canvasElement.width = this.width;
+        this.endPos = this.width - this.radius - this.lineWidth;
+        this.redrawCanvas();
+    }
+
+    /**
+     * Event gets called when mouse is moved out of the canvasElement, changes clicked variable to false to prevent drawing.
+     */
+    mouseLeave() {
+        this.clicked = false;
+    }
+
+    /**
+     * Event gets called when mouse is clicked, Starts the mouse drawing.
+     * @param  {number} $event Mouse Coordinates
+     */
+    mouseClick($event) {
+        this.setLineStyle('#9932cc', 10);
+        this.cx.beginPath();
+        this.tempStart.x = $event.clientX - this.canvasLeft;
+        this.tempStart.y = $event.clientY - this.canvasTop;
+        this.cx.moveTo(this.tempStart.x, this.tempStart.y);
+        this.clicked = true;
+    }
+
+    /**
+     * Event gets called when mouse is moved, traces mouse path if mouse is clicked,
+     * @param  {number} $event Mouse Coordinates
+     */
+    mousePos($event) {
+        if (this.clicked) {
```

Comment:
that code block is only run when `this.clicked` is `true`.
in other words, you could reduce the nested blocks if just return immediately when `this.clicked` is `false`.